reviews/models/mixins.py

```diff
@@ -0,0 +1,288 @@
+                raise ValueError('Preprint must have at least one subject to be published.')
+            self.reviewable.date_published = now
+            self.reviewable.is_published = True
+            enqueue_postcommit_task(get_and_set_preprint_identifiers, (), {'preprint': self.reviewable}, celery=True)
+        elif not should_publish and self.reviewable.is_published:
+            self.reviewable.is_published = False
+        self.reviewable.save()
+        node.save()
+
+    def resubmission_allowed(self, ev):
+        return self.reviewable.provider.reviews_workflow == workflow.Workflows.PRE_MODERATION.value
+
+    def notify_submit(self, ev):
+        context = self.get_context()
+        context['referrer'] = ev.kwargs.get('user')
+        user = ev.kwargs.get('user')
+        auth = Auth(user)
+        self.reviewable.node.add_log(
+            action=NodeLog.PREPRINT_INITIATED,
+            params={
+                'preprint': self.reviewable._id
+            },
+            auth=auth,
+            save=False,
+        )
+        reviews_signals.reviews_email_submit.send(context=context)
+
+    def notify_resubmit(self, ev):
+        context = self.get_context()
+        context['template'] = 'reviews_resubmission_confirmation'
+        reviews_signals.reviews_email.send(context=context)
+
+    def notify_accept_reject(self, ev):
+        context = self.get_context()
+        context['template'] = 'reviews_submission_status'
+        context['notify_comment'] = not self.reviewable.provider.reviews_comments_private and self.action.comment
+        context['is_rejected'] = self.action.to_state == workflow.States.REJECTED.value
+        context['was_pending'] = self.action.from_state == workflow.States.PENDING.value
+        reviews_signals.reviews_email.send(context=context)
+
+    def notify_edit_comment(self, ev):
+        context = self.get_context()
+        context['template'] = 'reviews_update_comment'
+        if not self.reviewable.provider.reviews_comments_private and self.action.comment:
+            reviews_signals.reviews_email.send(context=context)
+
+    def get_context(self):
+        return {
+            'domain': settings.DOMAIN,
+            'email_recipients': [contributor._id for contributor in self.reviewable.node.contributors],
+            'reviewable': self.reviewable,
+            'workflow': self.reviewable.provider.reviews_workflow,
+            'provider_url': self.reviewable.provider.domain or '{domain}preprints/{provider_id}'.format(domain=settings.DOMAIN, provider_id=self.reviewable.provider._id),
+            'provider_contact_email': self.reviewable.provider.email_contact or 'contact@osf.io',
+            'provider_support_email': self.reviewable.provider.email_support or 'support@osf.io',
+        }
+
+# Handle email notifications including: update comment, accept, and reject of submission.
+@reviews_signals.reviews_email.connect
+def reviews_notification(self, context):
```

Comment:
Note: This will be refactored in the 0.1.0 release to use a `mails.notify`-like function instead of checking subscriptions and creating `NotificationDigest`s manually.
https://openscience.atlassian.net/browse/MOD-150